velox/exec/tests/HashJoinTest.cpp

```diff
@@ -4669,6 +4670,87 @@ TEST_F(HashJoinTest, dynamicFiltersWithSkippedSplits) {
   }
 }
 
+TEST_F(HashJoinTest, dynamicFiltersAppliedToPreloadedSplits) {
+  vector_size_t size = 1000;
+  const int32_t numSplits = 5;
+  std::vector<RowVectorPtr> probeVectors;
+  probeVectors.reserve(numSplits);
+
+  std::vector<std::shared_ptr<TempFilePath>> tempFiles;
+  for (int32_t i = 0; i < numSplits; ++i) {
+    auto rowVector = makeRowVector({
+        makeFlatVector<int64_t>(
+            size, [&](auto row) { return (row + 1) * (i + 1); }),
+        makeFlatVector<int64_t>(size, [&](auto /*row*/) { return i; }),
+    });
+    probeVectors.push_back(rowVector);
+    tempFiles.push_back(TempFilePath::create());
+    writeToFile(tempFiles.back()->path, rowVector);
+  }
+  createDuckDbTable("u", probeVectors);
+
+  std::vector<RowVectorPtr> buildVectors{
+      makeRowVector({"c0"}, {makeFlatVector<int64_t>({0, numSplits})})};
+  createDuckDbTable("t", buildVectors);
+
+  auto makeInputSplits = [&](const core::PlanNodeId& nodeId) {
+    return [&] {
+      std::vector<exec::Split> probeSplits;
+      for (int32_t i = 0; i < numSplits; ++i) {
+        auto value = std::to_string(i);
+        auto split = facebook::velox::exec::test::HiveConnectorSplitBuilder(
+                         tempFiles[i]->path)
+                         .partitionKey("k", value)
+                         .build();
+        probeSplits.push_back(exec::Split(split));
+      }
+      SplitInput splits;
+      splits.emplace(nodeId, probeSplits);
+      return splits;
+    };
+  };
+  auto outputType =
+      ROW({"n1_0", "n1_1", "n1_2"}, {BIGINT(), BIGINT(), BIGINT()});
+  ColumnHandleMap assignments = {
+      {"n1_0", regularColumn("c0", BIGINT())},
+      {"n1_1", regularColumn("c1", BIGINT())},
+      {"n1_2", partitionKey("k", BIGINT())}};
```

Comment:
Is it important to use partition keys columns in this test? Would you explain why?

Reply:
Yes, we expect dynamic filters on the partition column to be generated. This way, we can verify whether the dynamic filter is passed to the preloaded data source by checking if the split is skipped during processing.